From f0a53588da7d6b642eeabdbd934d799ed0df2b8f Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Sun, 13 Feb 2022 18:46:23 -0800
Subject: [PATCH 074/127] x86-64: Define __memcmpeq in ld.so

Define __memcmpeq in ld.so so that compiler can generate __memcmpeq call
when compiling for ld.so.

(cherry picked from commit a5659cf27d3ce6101c1632715d18ab6321755340)
---
 sysdeps/x86_64/multiarch/memcmpeq-sse2.S | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sysdeps/x86_64/multiarch/memcmpeq-sse2.S b/sysdeps/x86_64/multiarch/memcmpeq-sse2.S
index b80a29d4b0..de7f5a7525 100644
--- a/sysdeps/x86_64/multiarch/memcmpeq-sse2.S
+++ b/sysdeps/x86_64/multiarch/memcmpeq-sse2.S
@@ -16,8 +16,10 @@
    License along with the GNU C Library; if not, see
    <https://www.gnu.org/licenses/>.  */
 
-#ifndef memcmp
+#if IS_IN (libc)
 # define memcmp	__memcmpeq_sse2
+#else
+# define memcmp	__memcmpeq
 #endif
 #define USE_AS_MEMCMPEQ	1
 #include "memcmp-sse2.S"
-- 
2.36.1

